sdk/storage/azure-storage-blobs/src/blob_client.cpp

```diff
@@ -531,6 +532,38 @@ namespace Azure { namespace Storage { namespace Blobs {
         *m_pipeline, m_blobUrl, protocolLayerOptions, context);
   }
 
+  Azure::Response<Models::CopyBlobFromUriResult> BlobClient::CopyFromUri(
+      const std::string& sourceUri,
+      const CopyBlobFromUriOptions& options,
+      const Azure::Core::Context& context) const
+  {
+    _detail::BlobRestClient::Blob::CopyBlobFromUriOptions protocolLayerOptions;
+    protocolLayerOptions.Metadata = options.Metadata;
+    protocolLayerOptions.Tags = options.Tags;
+    protocolLayerOptions.SourceUri = sourceUri;
+    protocolLayerOptions.AccessTier = options.AccessTier;
+    protocolLayerOptions.LeaseId = options.AccessConditions.LeaseId;
+    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
+    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
+    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
+    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
+    protocolLayerOptions.IfTags = options.AccessConditions.TagConditions;
```

Comment:
Tag condition seems to be not supported for this API.

Reply:
It's supported according to swagger https://github.com/Azure/azure-rest-api-specs/blob/2fbdd238b9fc19884e2d0491a338f7f6fc2aa1d7/specification/storage/data-plane/Microsoft.BlobStorage/preview/2020-02-10/blob.json#L6150

Reply:
and it's covered by unit test